Compiled with-nx.js imports from an unknown path

### Current Behavior

Recently this issue happened and was solved, but in the latest release it resurfaced. https://github.com/nrwl/nx/pull/15824

Running a built nextjs app errors out with the following error

```
node:internal/modules/cjs/loader:1075                                                                                                                                                                                                                                                                                                                     
  const err = new Error(message);                                                                                                                                                                                                                                                                                                                         
              ^                                                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                                                          
Error: Cannot find module '../src/utils/config'                                                                                                                                                                                                                                                                                                           
Require stack:                                                                                                                                                                                                                                                                                                                                            
- /usr/src/app/.nx-helpers/with-nx.js                                                                                                                                                                                                                                                                                                                     
- /usr/src/app/.nx-helpers/compiled.js                                                                                                                                                                                                                                                                                                                    
- /usr/src/app/next.config.js                                                                                                                                                                                                                                                                                                                             
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)                                                                                                                                                                                                                                                                                 
    at /usr/src/app/node_modules/.pnpm/next@13.4.1_@opentelemetry+api@1.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/require-hook.js:180:36                                                                                                                                                                                            
    at Module._load (node:internal/modules/cjs/loader:920:27)                                                                                                                                                                                                                                                                                             
    at Module.require (node:internal/modules/cjs/loader:1141:19)                                                                                                                                                                                                                                                                                          
    at require (node:internal/modules/cjs/helpers:110:18)                                                                                                                                                                                                                                                                                                 
    at Object.<anonymous> (/usr/src/app/.nx-helpers/with-nx.js:10:18)                                                                                                                                                                                                                                                                                     
    at Module._compile (node:internal/modules/cjs/loader:1254:14)                                                                                                                                                                                                                                                                                         
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)                                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1117:32)                                                                                                                                                                                                                                                                                             
    at Module._load (node:internal/modules/cjs/loader:958:12) {                                                                                                                                                                                                                                                                                           
  code: 'MODULE_NOT_FOUND',                                                                                                                                                                                                                                                                                                                               
  requireStack: [                                                                                                                                                                                                                                                                                                                                         
    '/usr/src/app/.nx-helpers/with-nx.js',                                                                                                                                                                                                                                                                                                                
    '/usr/src/app/.nx-helpers/compiled.js',                                                                                                                                                                                                                                                                                                               
    '/usr/src/app/next.config.js'                                                                                                                                                                                                                                                                                                                         
  ]                                                                                                                                                                                                                                                                                                                                                       
}   
```

Checking the js file on line 10 i get `const config_1 = require("../src/utils/config");` and config_1 is used twice in this kind of code

```
    (0, config_1.forNextVersion)('>=13.4.0', () => {
        process.env['__NEXT_PRIVATE_PREBUNDLED_REACT'] =
            // Not in Next 13.3 or earlier, so need to access config via string
            _nextConfig.experimental['serverActions'] ? 'experimental' : 'next';
    });
```

### Expected Behavior

This compiled version should not import files that are not there

### GitHub Repo

_No response_

### Steps to Reproduce

1.


### Nx Report

```shell
Node   : 18.16.0
   OS     : darwin arm64
   pnpm   : 8.4.0
   Hasher : Native
   
   nx                 : 16.1.1
   @nx/js             : 16.1.1
   @nx/jest           : 16.1.1
   @nx/linter         : 16.1.1
   @nx/workspace      : 16.1.1
   @nrwl/workspace    : 14.8.8
   @nx/cypress        : 16.1.1
   @nx/devkit         : 16.1.1
   @nrwl/devkit       : 14.8.8
   @nx/eslint-plugin  : 16.1.1
   @nx/nest           : 16.1.1
   @nx/next           : 16.1.1
   @nx/node           : 16.1.1
   @nx/react          : 16.1.1
   @nx/rollup         : 16.1.1
   @nrwl/tao          : 16.1.1
   @nx/web            : 16.1.1
   @nx/webpack        : 16.1.1
   nx-cloud           : 16.0.5
   typescript         : 5.0.4
   ---------------------------------------
   Community plugins:
   @nx-go/nx-go              : 2.7.0
   @nx-plus/docusaurus       : 14.1.0
   @nx-tools/nx-container    : 4.0.3
   @webpro/nx-tsc            : 0.0.1
   nest-commander-schematics : 3.0.0
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/workspace@14.8.8
     - @nrwl/devkit@14.8.8
   
   To fix this, run `nx migrate nx@16.1.1`
```


### Failure Logs

_No response_

### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/next/plugins/with-nx.ts
+++ b/packages/next/plugins/with-nx.ts
@@ -5,7 +5,6 @@
 import * as path from 'path';
 import type { NextConfig } from 'next';
 import type { NextConfigFn } from '../src/utils/config';
-import { forNextVersion } from '../src/utils/config';
 import type { NextBuildBuilderOptions } from '../src/utils/types';
 import type { DependentBuildableProjectNode } from '@nx/js/src/utils/buildable-libs-utils';
 import type { ProjectGraph, ProjectGraphProjectNode, Target } from '@nx/devkit';
@@ -463,6 +462,15 @@ export function getAliasForProject(
   }
 
   return null;
+}
+
+// Runs a function if the Next.js version satisfies the range.
+export function forNextVersion(range: string, fn: () => void) {
+  const semver = require('semver');
+  const nextJsVersion = require('next/package.json').version;
+  if (semver.satisfies(nextJsVersion, range)) {
+    fn();
+  }
 }
 
 // Support for older generated code: `const withNx = require('@nx/next/plugins/with-nx');`

--- a/packages/next/src/utils/config.ts
+++ b/packages/next/src/utils/config.ts
@@ -107,12 +107,3 @@ function isTsRule(r: RuleSetRule): boolean {
 
   return r.test.test('a.ts');
 }
-
-// Runs a function if the Next.js version satisfies the range.
-export function forNextVersion(range: string, fn: () => void) {
-  const semver = require('semver');
-  const nextJsVersion = require('next/package.json').version;
-  if (semver.satisfies(nextJsVersion, range)) {
-    fn();
-  }
-}

